src/js/node/http.ts

```diff
@@ -1664,8 +1677,12 @@ const OutgoingMessagePrototype = {
     headers.delete(name);
   },
 
-  setHeader(name, value) {
+  setHeader(name: string, value: unknown) {
+    if (this._header) {
+      throw $ERR_HTTP_HEADERS_SENT("set");
+    }
     validateHeaderName(name);
+    validateHeaderValue(name, value);
```

Comment:
yeah, setHeader should just do all of these validations including probably creating the header too.